src/Fisher_Kolmogorov_solver.cpp

```diff
@@ -0,0 +1,404 @@
+    DoFTools::make_sparsity_pattern(dof_handler, sparsity);
+    sparsity.compress();
+
+    pcout << "  Initializing the matrices" << std::endl;
+    jacobian_matrix.reinit(sparsity);
+
+    pcout << "  Initializing the system right-hand side" << std::endl;
+    residual_vector.reinit(locally_owned_dofs, MPI_COMM_WORLD);
+    pcout << "  Initializing the solution vector" << std::endl;
+    solution_owned.reinit(locally_owned_dofs, MPI_COMM_WORLD);
+    delta_owned.reinit(locally_owned_dofs, MPI_COMM_WORLD);
+
+    solution.reinit(locally_owned_dofs, locally_relevant_dofs, MPI_COMM_WORLD);
+    solution_old = solution;
+  }
+}
+ 
+template <int dim>
+void FisherKol<dim>::assemble_system()
+{
+  const unsigned int dofs_per_cell = fe->dofs_per_cell;
+  const unsigned int n_q           = quadrature->size();
+
+  FEValues<dim> fe_values(*fe,
+                          *quadrature,
+                          update_values | update_gradients |
+                            update_quadrature_points | update_JxW_values);
+
+  // Since we need to compute integrals on the boundary for Neumann conditions,
+  // we also need a FEValues object to compute quantities on boundary edges.
+  FEFaceValues<dim> fe_values_boundary(*fe,
+                                       *quadrature_boundary,
+                                       update_values |
+                                         update_quadrature_points |
+                                         update_JxW_values);
+
+  FullMatrix<double> cell_matrix(dofs_per_cell, dofs_per_cell);
+  Vector<double>     cell_residual(dofs_per_cell);
+
+  std::vector<types::global_dof_index> dof_indices(dofs_per_cell);
+
+  jacobian_matrix = 0.0;
+  residual_vector = 0.0;
+
+  // Value and gradient of the solution on current cell.
+  std::vector<double>         solution_loc(n_q);
+  std::vector<Tensor<1, dim>> solution_gradient_loc(n_q);
+
+  // Value of the solution at previous timestep (un) on current cell.
+  std::vector<double> solution_old_loc(n_q);
+
+  // The coefficients are constant throughout the program
+  const double alpha = parameters.get_double("coef_alpha");
+  const double deltat = parameters.get_double("deltat");
+
+  for (const auto &cell : dof_handler.active_cell_iterators())
+    {
+      if (!cell->is_locally_owned())
+        continue;
+
+      fe_values.reinit(cell);
+
+      cell_matrix   = 0.0;
+      cell_residual = 0.0;
+
+      fe_values.get_function_values(solution, solution_loc);
+      fe_values.get_function_gradients(solution, solution_gradient_loc);
+      fe_values.get_function_values(solution_old, solution_old_loc);
+
+      for (unsigned int q = 0; q < n_q; ++q)
+        {
+          Tensor<2, dim> D_matrix = fiber.value(fe_values.quadrature_point(q));
+
+          double f_loc = 0.0;
+          if (convergence_test)
+            f_loc = forcing_term_conv.value(fe_values.quadrature_point(q));
+          else 
+            f_loc = forcing_term_conv.value(fe_values.quadrature_point(q));
```

Comment:
It would seem that both branches are doing the same...
```suggestion
          const double f_loc = forcing_term_conv.value(fe_values.quadrature_point(q));
```